src/js/components/history/HistoryItem.tsx

```diff
@@ -0,0 +1,192 @@
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {View, parse, Spec} from 'vega';
+import {HistoryRecord, HistoryState} from '../../store/factory/History';
+import {cleanSpecForPreview} from '../../ctrl/demonstrations';
+import {startDragging, stopDragging} from '../../actions/inspectorActions';
+import {DraggingStateRecord, HistoryDraggingState} from '../../store/factory/Inspector';
+import sg from '../../ctrl/signals';
+import {MODE, SELECTED, CELL} from '../../store/factory/Signal';
+import {NumericValueRef, StringValueRef, tupleid} from 'vega';
+import {channelName} from '../../actions/bindChannel';
+import {setMarkVisual} from '../../actions/markActions';
+import * as vega from 'vega';
+import bindChannel from '../../actions/bindChannel';
+import {ColumnRecord, Schema} from '../../store/factory/Dataset';
+import { AnyAction } from 'redux';
+import {ThunkDispatch} from 'redux-thunk';
+import {State} from '../../store';
+import {setSignal} from '../../actions/signalActions';
+import {SignalValue} from 'vega-typings/types';
+import {batchGroupBy} from '../../reducers/historyOptions';
+
+const ctrl = require('../../ctrl');
+
+
+interface OwnProps {
+  id: string,
```

Comment:
consider passing `id` as a `number` if you call parseInt on it in all places where it's used